crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -499,6 +524,103 @@ impl<'a> TimelineEventHandler<'a> {
         self.state.reactions.map.insert(reaction_id, (reaction_sender_data, c.relates_to));
     }
 
+    #[instrument(skip_all, fields(replacement_event_id = ?replacement.event_id))]
+    fn handle_poll_start_edit(
+        &mut self,
+        replacement: Replacement<UnstablePollStartEventContentWithoutRelation>,
+    ) {
+        update_timeline_item!(self, &replacement.event_id, "poll edit", |event_item| {
+            if self.ctx.sender != event_item.sender() {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Edit event applies to another user's timeline item, discarding"
+                );
+                return None;
+            }
+
+            let poll_state = if let TimelineItemContent::Poll(poll_state) = &event_item.content() {
+                poll_state
+            } else {
+                info!(
+                    original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                    "Can't edit a poll that is not of type TimelineItemContent::Poll, discarding"
+                );
+                return None;
+            };
+
+            let new_content =
+                if let Ok(edited_poll_state) = poll_state.edit(&replacement.new_content) {
+                    TimelineItemContent::Poll(edited_poll_state)
+                } else {
+                    info!(
+                        original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
+                        "This poll can't be edited, discarding"
+                    );
+                    return None;
+                };
```

Comment:
```suggestion
            let new_content = match poll_state.edit(&replacement.new_content) {
                Ok(edited_poll_state) => TimelineItemContent::Poll(edited_poll_state),
                Err(e) => {
                    info!(
                        original_sender = ?event_item.sender(), edit_sender = ?self.ctx.sender,
                        "Failed to apply poll edit: {e}"
                    );
                    return None;
                }
            };
```

Reply:
Besides the different log message, why match better than if let in this context?

Reply:
Besides, had to use `{e:?}`.

Reply:
Being able to inspect the error is the only thing that really makes a difference, apart from formatting / indentation. However, I had missed that this is not using a proper error type (just `()`), so the if-let actually makes sense, there is no error message to log here.